Sealer/sealer.cc

```diff
@@ -603,38 +602,57 @@ void kRun(const ConnectPairsParams& params,
 
 	printLog(logStream, "Flanks inserted into k run = " + IntToString(flanks.size()) + "\n");
 
-	for (read1_it = flanks.begin(); read1_it != flanks.end();) {
-		success = false;
+	int counter = 0;
+	vector<map<FastaRecord, map<FastaRecord, Gap> >::iterator> flanks_closed;
+#pragma omp parallel private(read1_it, read2_it) firstprivate(counter)
+	for (read1_it = flanks.begin(); read1_it != flanks.end(); read1_it++) {
 		FastaRecord read1 = read1_it->first;
+		bool success = true;
 		for (read2_it = flanks[read1].begin(); read2_it != flanks[read1].end(); read2_it++) {
-			FastaRecord read2 = read2_it->first;
-
-			int startposition = read2_it->second.gapStart();
-			string tempSeq = merge(g, k, read2_it->second, read1, read2, params, g_count, traceStream);
-			if (!tempSeq.empty()) {
-				success = true;
-				allmerged[read1.id.substr(0,read1.id.length()-2)][startposition]
-					= ClosedGap(read2_it->second, tempSeq);
-//#pragma omp atomic
-				gapsclosed++;
-//#pragma omp atomic
-				uniqueGapsClosed++;
-				if (gapsclosed % 100 == 0)
-					printLog(logStream, IntToString(gapsclosed) + " gaps closed so far\n");
-
-				if (!opt::gapfilePath.empty()) {
-					gapStream << ">" << read1.id.substr(0,read1.id.length()-2)
-						  << "_" << read2_it->second.gapStart() << "-" << read2_it->second.gapEnd()
-						  << " LN:i:" << tempSeq.length() << '\n';
-					gapStream << tempSeq << '\n';
+			#ifdef _OPENMP
+			if (counter % omp_get_num_threads() == omp_get_thread_num()) {
```

Comment:
To avoid adding another level of indentation (and this function already has too many) I'd suggest instead…
```c++
if (counter % omp_get_num_threads() != omp_get_thread_num())
    continue;
```